File: python/algorithms/fibonacci_iterative.py
#!/usr/bin/env python


def fibonacci_iterative(number):
  """Returns the nth Fibonacci number in the Fibonacci sequence.

  The Fibonacci numbers are the numbers in the Fibonacci sequence, defined by
  the equation and seed values:

  F(n) = F(n - 1) + F(n - 2)

  F(0) = 0
  F(1) = 1

  * O(n) time complexity
  * O(1) space complexity

  Args:
    number: An integer.
  """
  if number == 0 or number == 1:
    return number
  a = 0
  b = 1
  for _ in range(2, number + 1):
    a, b = b, a + b
  return b

File: c/algorithms/fibonacci_iterative.c
#include "fibonacci_iterative.h"

unsigned int fibonacci_iterative(unsigned int number) {
  if (number == 0 || number == 1) {
    return number;
  }
  unsigned int a = 0;
  unsigned int b = 1;
  for (unsigned int i = 2; i <= number; ++i) {
    unsigned int c = a + b;
    a = b;
    b = c;
  }
  return b;
}
